apollo-audit/apollo-audit-impl/src/main/java/com/ctrip/framework/apollo/audit/context/ApolloAuditTracer.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.audit.context;
+
+import com.ctrip.framework.apollo.audit.annotation.OpType;
+import com.ctrip.framework.apollo.audit.constants.ApolloAuditHttpHeaderConstants;
+import com.ctrip.framework.apollo.audit.util.ApolloAuditUtil;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ApolloAuditTracer {
```

Comment:
From the design, 

a trace has many span

So we can use a stack in trace?

the span on the top of trace is activated,

and a trace can create a new span(follow or child)

Reply:
Actually, my design is similar to a stack, I put a reference "lastSpanContext" in AuditScope.
This is more like a linked list, each span with same parentSpan would link one by one.
Using stack will also met the problem that we can't confirm the new span should be the active span's son or sibling, because in my very first design, the follow-from relationship is only appearing in the siblings, the first child would follows from 'null'. So I don't think it is neccessary to use stack.
The very first design is more like the simple tree data structure "left−child right−sibling representation", while in my design it is left−parent.